core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -363,32 +392,10 @@ trait AttributeComponent {
       // delete attributes which are in exists but not save
       val attributesToDelete = existingAttrMap.filterKeys(! toSaveAttrMap.keySet.contains(_))
 
-      // update attributes which are in both to-save and currently-exists, but have different values.
-      // note that currently-existing attributes will have a populated id e.g. "1234", but to-save will have an id of "0"
-      // therefore, use a comparison function that looks at everything except id
-      // note this does not compare transactionId for AttributeScratchRecords. We do not expect AttributeScratchRecords
-      // here, and transactionId will eventually be going away, so don't bother
-      // TODO: should this compare function move closer to the AttributeRecord class?
-      def equalRecords(left: RECORD, right: RECORD): Boolean = {
-        // compare everything except id
-        left.name == right.name &&
-        left.valueString == right.valueString &&
-        left.valueNumber == right.valueNumber &&
-        left.valueBoolean == right.valueBoolean &&
-        left.valueJson == right.valueJson &&
-        left.valueEntityRef == right.valueEntityRef &&
-        left.listIndex == right.listIndex &&
-        left.listLength == right.listLength &&
-        left.namespace == right.namespace &&
-        left.ownerId == right.ownerId &&
-        left.deleted == right.deleted &&
-        left.deletedDate == right.deletedDate
-      }
-
       val attributesToUpdate = toSaveAttrMap.filter {
         case (k, v) =>
             existingKeys.contains(k) && // if the attribute doesn't already exist, don't attempt to update it
-            !existingAttributes.exists(equalRecords(_, v)) // if the attribute exists and is unchanged, don't update it
+            !existingAttributesSet.contains(ComparableRecord.fromRecord(v)) // if the attribute exists and is unchanged, don't update it
```

Comment:
1) let Scala compare case classes instead of manually inspecting the fields of the `RECORD` class
2) use `Set.contains` to look for existing attributes instead of `Traversable.exists`